### DISCUSSION TITLE
Dictation app from macOS based on whisper.

### DISCUSSION THREAD

**Discussion by foges** [openai/whisper#1128] [Show and tell]
I broke my arm over the weekend and was looking for a dictation app for Mac OS that was better than the built-in one. I didn't really find one I figured I'd build my own based on whisper. Have a look, would love your feedback. https://github.com/foges/whisper-dictation

**Comment by iloveitaly** [comment]
https://superwhisper.com looks like a great app!

**Reply by pradeepb28** [reply]
probably checkout https://audiowriter.app

**Reply by icreatenovelty** [reply]
Cool project. I find changing settings via terminal to be a little intimidating. FYI, [betterdictation.com](www.betterdictation.com) runs locally on M1+ macs and isn't a monthly subscription.

**Comment by SamMakesThings** [comment]
Any alternatives that will support an intel mac? SuperWhisper looks cool, but I'm not on silicon yet

**Comment by Diwannist** [comment]
Thank you for this tool. Is the AI model running MLX?

Could this app use an AI model that runs with MLX?

Could these models work?

https://pypi.org/project/mlx-whisper/

https://github.com/JosefAlbers/whisper-turbo-mlx

How does one confirm that this app is running an AI model on the GPU or CPU?

How do we go about implementing real-time features?

https://github.com/collabora/WhisperLive

Does this app do some sort of voice isolation AI work during or after each recording?

Does this app understand context?

Would it be better if we used a voice isolation AI model before we passed the audio to the transcriber model?

Is there a `language` parameter on the `transcribe` function in the traditional OpenAI Whisper package?

We made a new version of this app that runs on MLX Whisper.

https://github.com/computerstimulation/mlx-whisper-dictation

**Comment by pHequals7** [comment]
would love for folks to try [muesli](https://muesli.works) - it also runs on device on macOS/apple silicon - uses whisper, parakeet and bunch of other OSS models

**Comment by botaruibo** [comment]
Adding one more option for people who are specifically looking at **Chinese voice typing on macOS**.

I maintain **MyVoiceTyping**: https://github.com/botaruibo/MyVoiceTyping

It is not a Whisper-based project; the current local ASR path uses the FunASR / SenseVoice ecosystem, plus punctuation restoration and a small local GGUF text-polishing model. The goal is a little different from file transcription: hold a hotkey, speak Chinese, then paste cleaned-up text into the current app.

The main reason I’m sharing it in this thread is privacy/local-first usage. Audio and text can stay on the Mac by default, and the app, text-polishing model, and tuning dataset are all public:

- app: https://github.com/botaruibo/MyVoiceTyping
- model: https://modelscope.cn/models/botaruibo/MyVoiceTyping-1.5b-q4
- dataset: https://github.com/botaruibo/MyVoiceTyping-Dataset

Still early, but if anyone here is comparing local macOS dictation tools and cares about Chinese input + local text cleanup, feedback would be very welcome.
